### ISSUE TITLE
Missing item in the requirements list

### ISSUE THREAD

**Issue by avalentino** [pydata/numexpr#391]
Since ``numexpr` v2.8 ``setuptools`` is used in a couple modules, so it should be listed in the ``requirements.txt`` file.

More precisely, the vendored copy of ``packaging`` is actually used.
Probably it could make more sense to depend directly on ``packaging``.

If needed I can provide a PR for this.

**Comment by robbmcleod** [comment]
I'm a little unclear on the current state of packaging in Python in general. `distutils` is apparently deprecated. What is the replacement if not `setuptools`?

I don't see an issue with digging into the "protected" parts of `setuptools` that are deep into the package tree since there is functionality in `distutils` that `setuptools` does not natively have. Namely the `LooseVersion` method.

**Comment by emjohnson20** [comment]
I think this is related. The asammdf package is trying to install numexpr 2.8 and is failing with the error: 

```Downloading https://files.pythonhosted.org/packages/6f/18/4f07b0f5effb3b6fe4611738c15ce208f36dfe140c1f3ba1f28e62294717/numexpr-2.8.0.tar.gz (95kB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-3yr15e2w/numexpr/setup.py", line 16, in <module>
        import numpy as np
    ModuleNotFoundError: No module named 'numpy'```

**Comment by avalentino** [comment]
@robbmcleod not sure to understand your point.
Yes ``distutils`` is deprecated and ``setuptools`` is the natural replacement (not the only one).
I agree with you that using ``setuptools`` in the ``setup.py`` is the obvious solution at the moment.

Anyways I'm talking about the use of ``setuptools`` in the ``numbexpr`` package itself (namely ``expressions.py`` and ``test_numexpr.py``).
In this case you don't need complex features provided by ``setuptools``, you just need functions related to management of version parsing and comparison.
That functions are provided by the ``packaging`` standalone package that is considered as a reference implementation of some of the current standard for Python packaging and used by several tools to build more complex features.
``setuptools`` itself uses ``packaging``. The reason why they currently vendor it are not relevant in this context IMHO.

Also our use case perfectly matches one of the "Migration advice" provided in PEP632 [1] (``distutils`` deprecation).

[1] https://www.python.org/dev/peps/pep-0632/#migration-advice

In conclusion, leaving out philosophical considerations, the real point is that whenever we decide to use ``setuptools`` or ``packaging`` we should document that it is a runtime requirement.
Indeed we cannot assume that they are always installed since it is not true in several contexts, like e.g. conda packages and packages provided by GNU/Linux distributions like Debian/Ubuntu or Fedora/Redhat.

@emjohnson20 not sure that your problem is related to this specific issue.

You should use not use ``setup.py`` directly but rather the recommended
```shell
$ python3 -m pip install .
```
or
```shell
$ python3 -m build .
```

**Comment by robbmcleod** [comment]
I added both `setuptools` and `packaging` to `requirements.txt` and imported `packaging` directly in https://github.com/pydata/numexpr/commit/786ba03cf375948a17c7a505882a35ad5e6a407a.

**Comment by avalentino** [comment]
Thanks @robbmcleod. Please feel free to close this issue.

By the way  if you have ``packaging`` then it is not necessary to have also ``setuptools`` in the ``requirements.txt`` file.
As pointed out by @stephan-cr in #392 it would make more sense to have a ``pyproject.toml`` to list the build system dependencies.
Also for that I can provide a PR if you want.

**Comment by robbmcleod** [comment]
Ok, closing with release of 2.8.1, re-open if needed @avalentino .

**Comment by avalentino** [comment]
Perfect, thanks a lot @robbmcleod
